core/scripts.py

```diff
@@ -3563,6 +3563,66 @@ def _update_list_with_new_entries(self, custom_list: CustomList):
         custom_list.auto_update_status = CustomList.UPDATED
 
 
+class DeleteInvisibleLanesScript(LibraryInputScript):
+    """Delete lanes that are flagged as invisible including any visible sub-lanes."""
+
+    def process_library(self, library):
+
+        try:
+            continue_update = True
+            while continue_update:
+                rows = self._db.execute(
+                    "select count(l.id) from  lanes l, \
+                 (select id from lanes where visible = false and library_id = :library_id) invisible_lanes  \
+                 where l.visible = true and l.library_id = :library_id and  \
+                 l.parent_id = invisible_lanes.id",
+                    {"library_id": library.id},
+                )
+                if rows.first()[0] == 0:
+                    continue_update = False
+                else:
+                    self._db.execute(
+                        "update lanes set visible = false where id in ( select l.id from  lanes l, "
+                        "(select id from lanes where visible = false and  library_id = :library_id) "
+                        "invisible_lanes where l.visible = true and l.library_id = :library_id and  "
+                        "l.parent_id = invisible_lanes.id)",
+                        {"library_id": library.id},
+                    )
+
+            self._db.execute(
+                "delete from lanes_genres where lane_id in "
+                "(select id from lanes where library_id = :library_id and visible = false)",
+                {"library_id": library.id},
+            )
+
+            self._db.execute(
+                "delete from cachedfeeds where lane_id in ( "
+                "select id from lanes where library_id = :library_id and visible = false)",
+                {"library_id": library.id},
+            )
+
+            self._db.execute(
+                "delete from lanes where library_id = :library_id and visible = false",
+                {"library_id": library.id},
```

Comment:
Why raw SQL here instead of using the ORM?

Reply:
I can use the ORM if that is your preference.

Reply:
I'm more comfortable with raw SQL, but I can adapt.